canvas_modules/common-canvas/src/object-model/api-pipeline.js

```diff
@@ -936,6 +958,73 @@ export default class APIPipeline {
 		return movedNodesInfo;
 	}
 
+	// Returns an array of move link actions that can be used to reposition the
+	// links based on the provided Dagre output graph.
+	// Iterate nodes and check if there are any dummy nodes.
+	// If there are dummy nodes then links should have x/y cordinates as per dummy_nodes.
+	convertGraphToMoveLinks(outputGraph, canvasInfoPipelineLinks) {
+		const movedLinksInfo = {};
+		const linksPos = [];
+
+		const nodes = outputGraph.nodes;
+
+		nodes.forEach((node) => {
+			// Record the x and y position of dummy node to be assigned to the detached link.
+			// A fully detached link can have both temp-src-node and temp-trg-node
+			if (node.v.startsWith("temp-src-node-")) {
+				const dummyNodeId = node.v.split("temp-src-node-")[1];
+
+				linksPos[dummyNodeId] = {
+					...linksPos[node.v.split("temp-src-node-")[1]],
+					srcPos: {
+						x: node.value.x,
+						y: node.value.y
+					}
+				};
+			}
+			if (node.v.startsWith("temp-trg-node-")) {
+				const dummyNodeId = node.v.split("temp-trg-node-")[1];
+
+				linksPos[dummyNodeId] = {
+					...linksPos[node.v.split("temp-trg-node-")[1]],
+					trgPos: {
+						x: node.value.x,
+						y: node.value.y
+					}
+				};
+			}
+		});
+
```

Comment:
I don't think the code below is totally necessary. If you look in the `links.js` reducer at the code for case "SIZE_AND_POSITION_OBJECTS"  you'll see the only fields it needs for each element of `movedLinksInfo` is {id, srcPos, trgPos }.  So the  `movedLinksInfo` doesn't need to be an array of links with all their fields but just an array of elements with the above 3 fields which is basically what the linksPos array is.